inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.cpp

```diff
@@ -0,0 +1,516 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "mkldnn_strided_slice_node.h"
+
+#include <mkldnn_types.h>
+#include <mkldnn_extension_utils.h>
+
+#include "ie_parallel.hpp"
+#include <legacy/ie_layers.h>
+#include "common/dnnl_thread.hpp"
+#include "common/cpu_memcpy.h"
+#include "utils/general_utils.h"
+
+#include <string>
+#include <algorithm>
+
+#define THROW_ERROR THROW_IE_EXCEPTION << "StridedSlice layer with name '" << getName() << "' "
+
+using namespace mkldnn;
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+MKLDNNStridedSliceNode::MKLDNNStridedSliceNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache) :
+        MKLDNNNode(layer, eng, cache) {}
+
+void MKLDNNStridedSliceNode::getSupportedDescriptors() {
+    StridedSliceLayer* stridedSliceLayer = dynamic_cast<StridedSliceLayer*>(getCnnLayer().get());
+
+    if (stridedSliceLayer == nullptr)
+        THROW_ERROR << "cannot convert from CNN layer";
+
+    if (stridedSliceLayer->insData[DATA_ID].lock() == nullptr)
+        THROW_ERROR << "has nullable input data";
+
+    const SizeVector srcDims = stridedSliceLayer->insData[DATA_ID].lock()->getTensorDesc().getDims();
+    const SizeVector dstDims = stridedSliceLayer->outData[0]->getTensorDesc().getDims();
+
+    if (getParentEdges().size() < 3)
+        THROW_ERROR << "has incorrect number of input edges";
+    if (!getChildEdges().size())
+        THROW_ERROR << "has incorrect number of output edges";
+
+    SizeVector beginDims = stridedSliceLayer->insData[BEGIN_ID].lock()->getTensorDesc().getDims();
+    if (beginDims.size() != 1)
+        THROW_ERROR << " should have begin vector with 1 dimension";
+
+    SizeVector endDims = stridedSliceLayer->insData[END_ID].lock()->getTensorDesc().getDims();
+    if (endDims.size() > 1)
+        THROW_ERROR << "should have end vector with 1 dimension";
+    if (beginDims[0] != endDims[0])
+        THROW_ERROR << "should have begin vector with size equal to end vector size";
+
+    SizeVector strideDims;
+    if (stridedSliceLayer->insData.size() > 3) {
```

Comment:
stridedSliceLayer->insData.size() > STRIDE_ID

Reply:
Changed, thank you